### PR TITLE
Add error codes to reporter.panic in Contentful source plugin

### PR DESCRIPTION
**Pull request by sslotsky**
## Description

This adds error codes to `reporter.panic` calls that occur in `fetch.js`. Adding these codes helps us to categorize our build and preview failures.

### CODE REVIEW

**Comment by sslotsky** [comment]
Based on [this Slack thread with Lennart](https://gatsbyjs.slack.com/archives/C01AJ60GRUK/p1602752023054100) it may not be necessary to add this complexity. Called to `reporter.panic` can be restructured in a backwards compatible way by providing `sourceMessage` instead of `message` to the context.

File: packages/gatsby-source-contentful/src/report.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    text: context => context.sourceMessage,
+    level: `ERROR`,
+    category: `THIRD_PARTY`,
+  },
+  [CODES.FetchContentTypes]: {
+    text: context => context.sourceMessage,
+    level: `ERROR`,
+    category: `THIRD_PARTY`,
+  },
+}
```
Comment (sslotsky) [inline packages/gatsby-source-contentful/src/report.js:31]: This is where I most need feedback from the plugin authors. By using these codes with this error map, we want to keep the following things in mind:

1. Codes should be unique, meaningful, maintainable, and searchable.
1. The map should indicate the correct error category & level.

I made my best guess at assigning meaningful code names and categories and invite other contributors to offer corrections.

File: packages/gatsby-source-contentful/src/fetch.js [hunk 2]
```diff
@@ -48,12 +49,15 @@ module.exports = async function contentfulFetch({
...
-          `,`
-        )}' were found but were filtered down to none.`
-      )
+      reporter.panic({
+        id: CODES.LocalesMissing,
+        context: {
+          sourceMessage: `Please check if your localeFilter is configured properly. Locales '${_.join(
+            contentfulLocales.map(item => item.code),
+            `,`
+          )}' were found but were filtered down to none.`,
```
Comment (pvdz) [inline packages/gatsby-source-contentful/src/fetch.js:58]: No need for lodash here

```suggestion
          sourceMessage: `Please check if your localeFilter is configured properly. Locales '${
            contentfulLocales.map(item => item.code).join(`,`)
          }' were found but were filtered down to none.`,
```

File: packages/gatsby-source-contentful/src/report.js (hunk 1 above)
Comment (pvdz) [reply packages/gatsby-source-contentful/src/report.js:31]: I think this looks correct. I'm going to sign off on it but if @axe312ger has further thoughts we can fix forward.

**Review by pvdz — APPROVED** [review]
Just one minor change please.

**Pull request merged**